rmf_task/src/rmf_task/agv/TaskPlanner.cpp

```diff
@@ -370,6 +376,13 @@ struct Node
   double cost_estimate;
   rmf_traffic::Time latest_time;
   InvariantSet unassigned_invariants;
+  std::size_t next_available_internal_id = 1;
+
+  // ID 0 is reserved for charging tasks
+  std::size_t get_available_internal_id(bool charging_task = false)
+  {
+    return charging_task ? 0 : next_available_internal_id++;
```

Comment:
If we're generating new ids for all requests why not generate them for charging tasks too? That way each charging task has a unique id?

Reply:
It was so that the `Filter` does not differentiate between 2 assignments that have different IDs for a charging task, but are otherwise the same.

Reply:
Ah that makes sense. Thanks for the clarification.